File: c2-chaincode/chaincode-api/chaincode-api-fabric/src/main/java/io/komune/c2/chaincode/api/fabric/utils/FileUtils.java
package io.komune.c2.chaincode.api.fabric.utils;

import com.google.common.base.Strings;
import com.google.common.io.Resources;

import java.io.FileReader;
import java.net.MalformedURLException;
import java.net.URL;

public class FileUtils {

    public static final String FILE = "file:";

    public static URL getUrl(String resource) throws MalformedURLException {
        if(resource.startsWith(FILE)) {
            return new URL(resource);
        }
        return Resources.getResource(resource);
    }

    public static URL getUrl(String path, String resource) throws MalformedURLException {
        if (!Strings.isNullOrEmpty(path) && !path.endsWith("/")) {
            path = path + "/";
        }
        String fullPath = path + resource;
        return FileUtils.getUrl(fullPath);
    }

}

File: c2-chaincode/chaincode-api/chaincode-api-config/src/main/kotlin/io/komune/c2/chaincode/api/config/utils/FileUtils.kt
package io.komune.c2.chaincode.api.config.utils

import java.net.URL


object FileUtils {
    const val FILE: String = "file:"

    fun getUrl(path: String, resource: String): URL {
        val normalizedPath = if(!path.endsWith("/"))
            "$path/"
        else
            path
        val fullPath = normalizedPath + resource
        return getUrl(fullPath)
    }

    fun getUrl(resource: String): URL {
        if (resource.startsWith(FILE)) {
            return URL(resource)
        }
        return getResource(resource)
    }

    fun getResource(resourceName: String): URL {
        val url =  Thread.currentThread().contextClassLoader ?: FileUtils::class.java.getClassLoader()
        return url.getResource(resourceName)!!
    }
}
